drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -53,10 +53,15 @@ int intel_adsp_hda_dma_host_in_config(const struct device *dev,
 		*DGMBS(cfg->base, cfg->regblock_size, channel) =
 			blk_cfg->block_size & HDA_ALIGN_MASK;
 
-		if (dma_cfg->source_data_size <= 3) {
-			/* set the sample container set bit to 16bits */
+		/*
+		 * The Sample Container Size (SCS) bit
+		 * must be set to 0 for 32bit sample size (HD Audio container size)
+		 * otherwise it must be set to 1 for sample size container
+		 */
+		if (dma_cfg->source_data_size <= 3)
 			*DGCS(cfg->base, cfg->regblock_size, channel) |= DGCS_SCS;
-		}
+		else
+			*DGCS(cfg->base, cfg->regblock_size, channel) &= ~DGCS_SCS;
```

Comment:
the coding style requires braces around all conditions, same in the other ones

Reply:
Oh, I though the braces were there to not look silly with the comment, I'll add them in a minute

Reply:
I know, sorry, it's one of the gaps from the Linux codingstyle.

Reply:
no problem, I will remember from now on, thanks @fabiobaltieri for the reminder!